tests/rbd/test_rbd_compression.py

```diff
@@ -0,0 +1,222 @@
+    3. verify "rbd_compression_hint" to "compressible" on global, pool and image level
+    4. verify "rbd_compression_hint" to "incompressible" on global, pool and image level
+    5. Repeat the above steps for ecpool
+"""
+import json
+
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def validate_compression(self, mode, image_spec, **kw):
+    """
+    Method to validate after setting rbd_compression_hint
+    data is getting compressed or not.
+
+    Args:
+        mode: type of compression hints (compressible or incompressible)
+        image_spec: poolname + imagename
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    config = kw.get("config")
+    io = config.get("io_total")
+    kw["io"] = io
+    kw["imagespec"] = image_spec
+    pool_name = image_spec.split("/")[0]
+
+    # running io to check compression of data
+    self.rbd_bench(**kw)
+
+    out = self.exec_cmd(cmd="ceph df detail --format json-pretty", output=True)
+    output = json.loads(out.rstrip())
+    pool_info = output["pools"]
+
+    # Iterate over the pools to get compressed data information
+    for pool in pool_info:
+        if pool["name"] == pool_name:
+            compressed_data = pool["stats"]["compress_bytes_used"]
+            break
+    log.debug(pool["stats"])
+    if mode == "compressible":
+        if compressed_data == 0:
+            log.error("Test Failed: Data did not get compressed in compressible mode")
+            return 1
+        else:
+            log.info(
+                f"Test Passed: Data got compressed, bytes compressed: {compressed_data}"
+            )
+            return 0
+
+    elif mode == "incompressible":
+        if compressed_data != 0:
+            log.error("Test Failed: Data gets compressed in incompressible mode")
+            return 1
+        else:
+            log.info(
+                "Test Passed: Data did not get compressed due to incompressible mode set"
+            )
+            return 0
+    else:
+        log.error(f"Invalid mode: {mode}")
+        return 1
+
+
+def test_compression(rbd, pool_type, **kw):
+    """
+    Run rbd_compression_hint test on supported options,
+    i.e compressible and incompressible.
+
+    Args:
+        rbd: RBD object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    config = kw.get("config")
+
+    compression_algorithm = config.get("compression_algorithm")
+    compression_mode = config.get("compression_mode")
+    compression_ratio = config.get("compression_ratio")
+
+    pool_name = config[pool_type]["pool"]
+
+    compression_configs = {
+        "compression_algorithm": compression_algorithm,
+        "compression_mode": compression_mode,
+        "compression_required_ratio": compression_ratio,
+    }
```

Comment:
If we are not using compression_ratio/compression_mode/compression_algorithm after using it to initiate compression_configs, why do we need them ?
In other words, can we merge 
line 97-99 with 104-106

Reply:
thanks for the code optimization point, Done